cpp/src/arrow/csv/reader.cc

```diff
@@ -307,7 +309,7 @@ class ThreadedBlockReader : public BlockReader {
       RETURN_NOT_OK(chunker_->ProcessSkip(current_partial, current_buffer, is_final,
                                           &skip_rows_, &current_buffer));
       bytes_skipped += orig_size - current_buffer->size();
-      current_partial = SliceBuffer(current_buffer, 0, 0);
+      current_partial = std::make_shared<Buffer>(nullptr, 0);
```

Comment:
Won't this have the same problem for the threaded reader?

Reply:
No because in the block right below this `partial_` is correctly set `partial_ = std::move(current_buffer);`. The bug only was in the serial reader.

Reply:
Yep.  I see it now.